### PR TITLE
Update cli.py usability

### LINKED ISSUE #4: Straight forward README
Update readme to give straight forward info on how to run a target KNoT app and how to use the cli.py tool.

### PR DESCRIPTION
**Pull request by jcorde**
General enhancements for command line interface given at `scripts/cli.py` that increase usability, mainly focused on the board nrf52840_pca10059 (Dongle).
Extra support was also added.

- Add set default target board option using an alias or board name.
- Add set default OpenThread path used at compilation.
- Check if the signed image is present before trying to flash it.
- Add `flash` command that permits flashing without recompiling.
- Add `make menuconfig` option.
- Set `--flash` and `--clear` as an option for `make`.
- Select mcuboot stock image based on the target board.
- Add option to flash merged mcuboot and app image.
- Flash Dongle with automatic target board port detection using Open DFU Bootloader.
- List supported boards and their aliases.
- Fix app size at signature based on the target board.
- Update Readme to support the above changes.

Fixes #4.

### CODE REVIEW

File: scripts/cli.py [hunk 1]
```diff
@@ -247,6 +250,31 @@ def set_quiet(self, quiet):
...
+        """
+        # Get ports that ma
+        re = self.Constants.USB_HID_RE.format(vid, pid)
+        ports = list(serial.tools.list_ports.grep(re))
+
+        # Abort if no board found
+        if not ports:
+            logging.critical('Error: No target board connected!')
+            logging.info('Make sure the board is connected and programable')
+            exit()
```
Comment (ramonhpr) [inline scripts/cli.py:304]: Provide  exit number less than 0 when there is error

File: scripts/cli.py [hunk 2]
```diff
@@ -247,6 +250,31 @@ def set_quiet(self, quiet):
...
+
+        # Abort if more than one boards found
+        if len(ports) > 1:
+            logging.critical('Error: Too many target boards connected!')
+            logging.info("Provide the target device with '--port <PORT>' \
+or remove the other boards")
+            exit()
```
Comment (ramonhpr) [inline scripts/cli.py:311]: same here

File: scripts/cli.py [hunk 3]
```diff
@@ -247,6 +250,31 @@ def set_quiet(self, quiet):
     def set_debug(self, debug):
         self.debug_app = debug
 
+    def find_flash_dev(self, vid, pid):
+        """
+        Return device port based on Vendor Id and Product Id.
+        Abort if no board or more than one found.
+        """
+        # Get ports that ma
```
Comment (ramonhpr) [inline scripts/cli.py]: Is there some missing comment here?

**Review by ramonhpr — CHANGES_REQUESTED** [review]
There is more than 72 columns in this commit message body
hash: 078fe9b4f66f8347f5262c928895a1a6cf92bcbd

File: scripts/cli.py [hunk 4]
```diff
@@ -326,21 +330,26 @@ def make_core_dir(self):
...
-        cmd = 'mergehex -m {} {} -o {}'.format(self.setup_app.hex_path,
-                                               self.main_app.hex_path,
-                                               self.merged_hex_path)
-        logging.info('Merging main and setup apps')
+        cmd = 'mergehex -m {} {} -o {}'.format(input_1, input_2, output)
         run_cmd(cmd)
-        if not os.path.isfile(self.merged_hex_path):
-            exit('Error: No hex file found at {}'.format(self.merged_hex_path))
+        if not os.path.isfile(output):
+            exit('Error: No hex file found at {}'.format(output))
```
Comment (ramonhpr) [inline scripts/cli.py]: use `logging.critical`

File: scripts/cli.py [hunk 5]
```diff
@@ -491,6 +491,11 @@ def mcuboot():
     KnotSDK().flash_mcuboot()
 
 
+@cli.command(help='Flash Setup and Main apps')
+def flash():
+    KnotSDK().flash_signed()
+
```
Comment (ramonhpr) [inline scripts/cli.py]: just one breaking line

Comment (ramonhpr) [inline scripts/cli.py:720]: just one breaking line

File: scripts/cli.py (hunk 4 above)
Comment (ramonhpr) [inline scripts/cli.py]: use `logging.critical`

File: scripts/cli.py [hunk 6]
```diff
@@ -491,6 +491,13 @@ def set():
 def board(board):
     Config().set(Config().KEY_BOARD, board)
 
+
```
Comment (ramonhpr) [inline scripts/cli.py:741]: On commit message: 
`Add 'set ot-path <OT-PATH>' command option that permits the user to` -> changes `permits` to `allows`
`The defined value will be saved at the config file alongside with with` -> Use just one `with`

**Review by ramonhpr — CHANGES_REQUESTED** [review]

File: scripts/cli.py (hunk 5 above)
Comment (jcorde) [reply scripts/cli.py:720]: As this is a top level function declaration, there should be two blank lines between the declarations.
https://www.python.org/dev/peps/pep-0008/#blank-lines

Comment (ramonhpr) [reply scripts/cli.py:720]: Ack

File: scripts/cli.py (hunk 6 above)
Comment (jcorde) [reply scripts/cli.py:741]: Fixed.
Thank you!

File: scripts/cli.py (hunk 4 above)
Comment (jcorde) [reply scripts/cli.py]: Done.
Thank you!

File: scripts/cli.py (hunk 3 above)
Comment (jcorde) [reply scripts/cli.py]: Indeed. Fixed!
Thank you!

File: scripts/cli.py (hunk 1 above)
Comment (jcorde) [reply scripts/cli.py:304]: I totally agree with you. But, for a matter of pattern keeping, we agreed to keep this behavior and fix all its occurrences in a single future commit.

File: scripts/cli.py (hunk 2 above)
Comment (jcorde) [reply scripts/cli.py:311]: Same as above. Thank you for noticing!

**Review by ramonhpr — APPROVED** [review]
Looks good to me!

**Pull request merged**